### PR TITLE
MCR-2192 mycore csl

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2192).

### CODE REVIEW

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-csl/src/main/java/org/mycore/csl/MCRCSLTransformerInstance.java  3
- mycore-csl/src/main/java/org/mycore/csl/MCRItemDataProvider.java  3
- mycore-csl/src/main/java/org/mycore/csl/MCRBibTeXCSLTransformer.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5401931&bid=17708596)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 782879a4dd5babe3959437c267caeca40b358c0b! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5436-pull_request-953

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 782879a4dd5babe3959437c267caeca40b358c0b! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5436-pull_request-953

**Review by fluetze — APPROVED** [review]
Looks good. Thx!

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 782879a4dd5babe3959437c267caeca40b358c0b! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5436-pull_request-953

**Comment by kkrebs** [comment]
please fix build errors:

[WARNING] Unused declared dependencies found:
[WARNING]    org.citationstyles:locales:jar:2.0.0:compile
[WARNING]    org.citationstyles:styles:jar:2.0.0:compile

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.1.1:analyze-only (analyze) on project mycore-csl: Dependency problems found -> [Help 1]
[ERROR]

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 782879a4dd5babe3959437c267caeca40b358c0b! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5436-pull_request-953

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 782879a4dd5babe3959437c267caeca40b358c0b! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5436-pull_request-953

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on ae3f3d246f8a820520ca8ae4313df979d7e98fc9! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5446-pull_request-953

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-csl/src/main/java/org/mycore/csl/MCRCSLTransformerInstance.java  3
- mycore-csl/src/main/java/org/mycore/csl/MCRBibTeXCSLTransformer.java  7
- mycore-csl/src/main/java/org/mycore/csl/MCRItemDataProvider.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5401931&bid=17708596)

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-csl/src/main/java/org/mycore/csl/MCRBibTeXCSLTransformer.java  7
- mycore-csl/src/main/java/org/mycore/csl/MCRCSLTransformerInstance.java  3
- mycore-csl/src/main/java/org/mycore/csl/MCRItemDataProvider.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5401931&bid=17708596)

File: mycore-csl/src/main/java/org/mycore/csl/MCRBibTeXCSLTransformer.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                format);
+            if (!styleFormatTransformerStack.contains(instance)) {
+                styleFormatTransformerStack.push(instance);
+            }
+        }
+    }
+
+    @Override public MCRContent transform(MCRContent bibtext, MCRParameterCollector parameter) throws IOException {
+        final String format = parameter!=null? parameter.getParameter("format", DEFAULT_FORMAT): DEFAULT_FORMAT;
+        final String style =  parameter!=null? parameter.getParameter("style", DEFAULT_STYLE): DEFAULT_STYLE;
```
Comment (yagee-de) [inline mycore-csl/src/main/java/org/mycore/csl/MCRBibTeXCSLTransformer.java]: please check code style

File: mycore-csl/src/main/resources/components/csl/config/mycore.properties [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# MCR.ContentTransformer.mods2bibmods.Class=org.mycore.common.content.transformer.MCRXSLTransformer
+# MCR.ContentTransformer.mods2bibmods.Stylesheet=xsl/mycoreobject-mods.xsl,xsl/mods2bibmods.xsl
+# MCR.ContentTransformer.mods2bibmods.MIMEType=text/xml
+# MCR.ContentTransformer.mods2bibmods.FileExtension=xml
+
+# MCR.ContentTransformer.mods2bibtex.Class=org.mycore.common.content.transformer.MCRBibUtilsTransformer
+# local.property: system command for xml2bib
+# MCR.ContentTransformer.mods2bibtex.Command=%UBO.BibUtils.Base%xml2bib -b -w -o UTF8
```
Comment (yagee-de) [inline mycore-csl/src/main/resources/components/csl/config/mycore.properties]: Do not use properties that are not defined elsewhere

Comment (yagee-de) [reply mycore-csl/src/main/resources/components/csl/config/mycore.properties]: How about removing properties from mycore-mods and add a comment:
#requires runtime dependency mycore-mods
and including the last 3 lines as active properties

**Review by yagee-de — CHANGES_REQUESTED** [review]

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-csl/src/main/java/org/mycore/csl/MCRBibTeXCSLTransformer.java  7
- mycore-csl/src/main/java/org/mycore/csl/MCRItemDataProvider.java  3
- mycore-csl/src/main/java/org/mycore/csl/MCRCSLTransformerInstance.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5401931&bid=17708596)

**Pull request merged**